sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/models/AvailabilityData.java

```diff
@@ -223,4 +226,99 @@ public AvailabilityData setMeasurements(Map<String, Double> measurements) {
         this.measurements = measurements;
         return this;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AvailabilityData setVersion(int version) {
+        super.setVersion(version);
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        jsonWriter.writeStartObject();
+        jsonWriter.writeIntField("ver", getVersion());
+        jsonWriter.writeStringField("id", this.id);
+        jsonWriter.writeStringField("name", this.name);
+        jsonWriter.writeStringField("duration", this.duration);
+        jsonWriter.writeBooleanField("success", this.success);
+        jsonWriter.writeStringField("runLocation", this.runLocation);
+        jsonWriter.writeStringField("message", this.message);
+        writeMap(this.properties, "properties", jsonWriter, (element) -> {
+            try {
+                jsonWriter.writeString(element);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
can we simplify, since this is our `writeMap` class, we could change the 4th arg to be something that allows us to throw IOException?
```suggestion
            jsonWriter.writeString(element);
```

Reply:
this makes intellij happy, it's inside a function even i have the method signature to throws

Reply:
Is the interface being used as the callback defines `throws IOException` on the lambda method, a try-catch here isn't needed. I'd recommend changing the 4th arg to be `WriteValueCallback` as this design mimics the purpose of that interface.